commit f6fca6f4d4241d2624759734c6b91e42e4f7bf22
Author: Patrick Wildt <patrick@blueri.se>
Date:   Sat May 2 21:12:49 2015 +0200

    llvm: disable NEON on ARM targets
    
    The Marvell Armada XP does not have NEON/SIMD. Our default target
    defaults to using NEON, so we have to disable this.
    
    It's more of a workaround, I'm not happy with it, but there's not
    really a better way.

diff --git a/contrib/llvm/tools/clang/lib/Driver/Tools.cpp b/contrib/llvm/tools/clang/lib/Driver/Tools.cpp
index a1c6bee..e69c8c2 100644
--- a/contrib/llvm/tools/clang/lib/Driver/Tools.cpp
+++ b/contrib/llvm/tools/clang/lib/Driver/Tools.cpp
@@ -730,6 +730,13 @@ static void getARMTargetFeatures(const Driver &D, const llvm::Triple &Triple,
       Features.push_back("+soft-float-abi");
   }
 
+  /* Disable NEON on Bitrig unless overridden. */
+  if (Triple.getOS() == llvm::Triple::Bitrig &&
+      !Args.getLastArg(options::OPT_mfpu_EQ)) {
+    Features.push_back("+vfp3");
+    Features.push_back("-neon");
+  }
+
   // Honor -mfpu=.
   if (const Arg *A = Args.getLastArg(options::OPT_mfpu_EQ))
     getARMFPUFeatures(D, A, Args, Features);
